web/src/components/vis/ScorePlotTile.vue

```diff
@@ -44,6 +48,32 @@ export default {
     pcY() {
       return Number.parseInt(this.pcYval, 10);
     },
+
+    pcCoords() {
+      return this.maybeData('x', []);
+    },
+
+    rowLabels() {
+      return this.maybeData('rows', []);
+    },
+
+    groupLabels() {
+      return this.maybeData('labels', {});
+    },
+
+    eigenvalues() {
+      return this.maybeData('sdev', []);
+    },
+  },
+
+  methods: {
+    maybeData(key, dflt) {
+      const {
+        plot,
+      } = this;
+
+      return plot.data ? plot.data[key] : dflt;
```

Comment:
I do this a lot myself, but it's technically not correct.  If your actual value is falsey and your default value is truthy, this will produce the wrong output.  Better to check if `plot.data[key] === undefined`.

I used to be more rigorous about it, then I settled into using this and `value || default`.

Reply:
I might be missing something: this checks to see whether `plot.data` is falsy, not `plot.data[key]`. My understanding is that if the plot data isn't ready yet, `plot.data === null` holds, and this method would therefore return `dflt`. If `plot.data` has an actual object in it, then `plot.data[key]` comes back as the result, regardless of its truthiness. No?

Reply:
Yep, you're right.

Reply:
(I am still curious about this topic in general, but it's mooted by using the plotData mixin's data grabbing method.)